calliope/cli.py

```diff
@@ -172,23 +193,63 @@ def new(path, template, debug):
         shutil.copytree(source_path, path)
 
 
-@cli.command(short_help='Run a model.')
+def _run_setup_model(
+        model_file, scenario, model_format, override_dict):
+    """
+    Build model in CLI commands. Returns ``model``, a ready-to-run
+    calliope.Model instance.
+
+    """
+    # Try to determine model file type if not given explicitly
+    if model_format is None:
+        if model_file.split('.')[-1] in ['yaml', 'yml']:
+            model_format = 'yaml'
+        elif model_file.split('.')[-1] in ['nc', 'nc4', 'netcdf']:
+            model_format = 'netcdf'
+        else:
+            raise ValueError(
+                'Cannot determine model file format based on file '
+                'extension for "{}". Set format explicitly with '
+                '--model_format.'.format(model_file)
+            )
+
+    if model_format == 'yaml':
+        model = Model(
+            model_file, scenario=scenario, override_dict=override_dict
+        )
+    elif model_format == 'netcdf':
+        if scenario is not None or override_dict is not None:
+            raise ValueError(
+                'When loading a pre-built model from NetCDF, the '
+                '--scenario and --override_dict options are not available.'
+            )
+        model = read_netcdf(model_file)
+    else:
+        raise ValueError('Invalid model format: {}'.format(model_format))
+
+    return model
+
+
+@cli.command(short_help='Build and run a model.')
 @click.argument('model_file')
 @click.option('--scenario')
+@click.option('--model_format')
+@click.option('--override_dict')
 @click.option('--save_netcdf')
 @click.option('--save_csv')
 @click.option('--save_plots')
 @click.option('--save_logs')
-@click.option('--model_format')
-@click.option('--override_dict')
+@click.option(
```

Comment:
Any reason to not put in the help strings for the other arguments?

Reply:
No, but these were already missing before, and this PR already does a lot of stuff ;)